File: Python/milkpails.py
import sys
sys.stdin = open("pails.in", "r")
sys.stdout = open("pails.out", "w")
x, y, m = sys.stdin.readline().strip().split()
x, y, m = int(x), int(y), int(m)
maxAmount = 0
for i in range((m//x)+1):
    for e in range((m//y)+1):
        amount = (i*x)+(e*y)
        if amount <= m:
            maxAmount = max(maxAmount, amount)
print(maxAmount)

File: CPP/milkpails.cpp
#include <iostream>
using namespace std;

int main() {
    freopen("pails.in", "r", stdin);
    freopen("pails.out", "w", stdout);
    int x, y, m;
    cin >> x >> y >> m;
    int thing{m/x};
    int max{};
    for (int i{0}; i < thing+1; i++) {
        for (int j{0}; j < thing+1; j++) {
            int added{(i*x)+(j*y)};
            if (added>m) {
                break;
            }
            if (added > max){max = added;}}
        }
    cout << max;
    }